servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/ChannelSet.java

```diff
@@ -146,29 +146,47 @@ protected void handleSubscribe(final Subscriber subscriber) {
                 for (final Channel channel : channelMap.values()) {
                     Attribute<AsyncCloseable> closeableAttribute = channel.attr(CHANNEL_CLOSEABLE_KEY);
                     AsyncCloseable channelCloseable = closeableAttribute.getAndSet(null);
-                    if (channelCloseable != null) {
-                        // Upon shutdown of the set, we will close all live channels. If close of individual channels
-                        // are offloaded, then this would trigger a surge in threads required to offload these closures.
-                        // Here we assume that if there is any offloading required, it is done by offloading the
-                        // Completable returned by closeAsyncGracefully() hence offloading each channel is not required.
-                        // Hence, we override the offloading on each channel for this particular subscribe to use
-                        // immediate and effectively disable offloading on each channel.
-                        closeable.merge(new AsyncCloseable() {
-                            @Override
-                            public Completable closeAsync() {
-                                return channelCloseable.closeAsync().publishOnOverride(immediate());
-                            }
-
-                            @Override
-                            public Completable closeAsyncGracefully() {
-                                return channelCloseable.closeAsyncGracefully().publishOnOverride(immediate());
-                            }
-                        });
+                    if (null != channelCloseable) {
+                        // Upon shutdown of the set, we will close all live channels. If close of individual
+                        // channels are offloaded, then this would trigger a surge in threads required to offload
+                        // these closures.
+                        if (channelCloseable instanceof NettyChannelListenableAsyncCloseable) {
```

Comment:
Because this is an internal code, should we change the type of `CHANNEL_CLOSEABLE_KEY` to `NettyChannelListenableAsyncCloseable` and never worry about another type here?

Reply:
Yes, I think we should use only `NettyChannelListenableAsyncCloseable` for the `CHANNEL_CLOSEABLE_KEY`. Everything works, only mock types in `ChannelSetTest` need to be updated.